zebra/zebra_rnh.c

```diff
@@ -597,8 +601,8 @@ static int check_overlay_nexthop(struct prefix *pp, uint8_t *isreachable)
 	}
 
 	//get the nextop to check if it is connected. 
-	inet_ntop(pp->family, &pp->u.prefix, via, PREFIX2STR_BUFFER);
-	snprintf(cntrname, 256, "nh.%s", via);
+	inet_ntop(pp->family, &pp->u.prefix, nh, PREFIX2STR_BUFFER);
+	snprintf(cntrname, 256, "nh.%s", nh);
```

Comment:
:exclamation:**Cycode**: SAST violation: 'Locate all calls to interesting and potentially insecure API functions (candidate points). The auditor can backtrace from these candidate points to find pathways allowing access from untrusted input.'.
 
Severity: Medium
 <!-- d2a0312f-4ee0-4dfd-a1de-bc1f1143aeee -->